File: LAB2_C++/LAB2_C++/String_Class.cpp
#include "String_Class.h"
#include <string>

String_Class::String_Class() //Конструктор за замовчуванням
{
    value = " ";
}

String_Class::String_Class(std::string str) //Конструктор с параметром
{
    value = str;
}

String_Class::String_Class(const String_Class& other_value) //Конструктор копіювання
{
    value = other_value.value;
}

std::string String_Class::Get_Word()
{
    return value;
}

void String_Class::Set_Word(const std::string str)
{
    String_Class::value = str;
}


int String_Class::Get_Length()
{
    return value.length();
}

void String_Class::Sort_Descending()
{
    int n = value.length();

    for (int i = 0; i < n - 1; i++)
    {
        for (int j = 0; j < n - i - 1; j++)
        {
            if (value[j] < value[j + 1])
            {
                char temp = value[j];
                value[j] = value[j + 1];
                value[j + 1] = temp;
            }

        }
    }
}

String_Class::~String_Class()
{

}

File: LAB2_C#/LAB2_C#/String_Class.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace LAB2_C_
{
    internal class String_Class
    {
        private string value;

        public String_Class() //Конструктор за замовчуванням
        {
            value = " ";

        }

        public String_Class(string str) //Конструктор с параметром 
        {
            str = "Aviation";
            value = str;
        }
        
        public String_Class(String_Class other_value) //Конструктор копіювання
        {
            value = other_value.value;
       }

        public void Set_Word(string str)
        {
            value = str;
        }
        public string Get_Word()
        {
            return value;
        }

        public int Get_Length()
        {
            return value.Length;
        }

        public void Sort_Descending()
        {
            char[] arr = value.ToCharArray();
            Array.Sort(arr);
            Array.Reverse(arr);
            value = new string(arr);
        }
        ~String_Class()
        {
            
        }

    }
}
